cmd/terminus/main.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"context"
+	dbsql "database/sql"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"regexp"
+	"syscall"
+
+	"github.com/treeverse/terminus/pkg/http"
+	"github.com/treeverse/terminus/pkg/queue_handler"
+	"github.com/treeverse/terminus/pkg/store/sql"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/dustin/go-humanize"
+	_ "github.com/jackc/pgx/v4/stdlib"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func main() {
+	Execute()
+}
+
+func DieOnErr(err error) {
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func NewSQS() (*sqs.SQS, error) {
+	sess, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("New AWS session: %w", err)
+	}
+	sqs := sqs.New(sess)
+	return sqs, nil
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "terminus",
+	Short: "Terminus monitors and optionally controls quotas for lakeFS users on S3",
+	Long: `Terminus is a service that listens to S3 events in order to track bytes used.
+It may can track S3 resources used by lakeFS installations or users.`,
+}
+
+// GetFlagBytes returns the value of flag using humanized bytes.  E.g. "8K" -> 8192.
+func GetFlagBytes(flags *pflag.FlagSet, flag string) (int64, error) {
+	s, err := flags.GetString(flag)
+	if err != nil {
+		return 0, fmt.Errorf("get flag %s: %w", flag, err)
+	}
+	bytes, err := humanize.ParseBytes(s)
+	if err != nil {
+		return 0, fmt.Errorf("parse flag %s value %s: %w", flag, s)
+	}
+	if int64(bytes) < 0 {
```

Comment:
Question - Is that a reasonable limit to use? Shouldn't we use a reasonable max quota here?

Reply:
There is no max on S3.  2^63 bytes is 8 EiB or 8192 PiB.  So we won't need that much in the foreseeable future.

Going beyond that, this code is anyway broken at the 63/64 bit limit.  E.g. we hold usage as a PostgreSQL BIGINT, which _is_ `int64`.  We'd need to represent bigger ints across the board -- out of scope.